### PR TITLE
MAU-73 - Allow user to make components Optional or mandatory

### PR DESCRIPTION
**Pull request by john-coxon**

### CODE REVIEW

File: src/components/Templates/TemplateDragContainer.js [hunk 1]
```diff
@@ -19,23 +21,41 @@ export default class TemplateDragContainer extends Component {
...
     const { htmlString, onChange } = this.props;
-    let commentCount = 0;
-    let newsfeedCount = 0;
     const parser = new DOMParser();
-    const commentReplacer = () => {
-      ++commentCount;
-      return `Comment ${commentCount}`;
+    const typeRegister = {};
+    const typeReplacer = (fieldType) => {
+      if (typeof typeRegister[fieldType] === 'undefined') {
```
Comment (snozza) [inline src/components/Templates/TemplateDragContainer.js]: Can this just be  if (!typeRegister[fieldType]) {

File: src/components/Validator/validator.js [hunk 2]
```diff
@@ -1,8 +1,31 @@
 import validator from 'tv4';
+import { markDownPattern } from '../../utils/regex';
+const cheerio = require('cheerio');
 
-validator.setErrorReporter((error, data, fieldSchema) => {
-  const { description, message = 'is required' } = fieldSchema;
-  return `${description} ${message}`;
+validator.setErrorReporter((error, data) => {
```
Comment (snozza) [inline src/components/Validator/validator.js]: What are the possible types of data? other than object

File: src/components/Templates/TemplateDragContainer.js (hunk 1 above)
Comment (john-coxon) [inline src/components/Templates/TemplateDragContainer.js]: Yes, now changed

File: src/components/Validator/validator.js (hunk 2 above)
Comment (john-coxon) [inline src/components/Validator/validator.js]: String

**Pull request merged**